### PR TITLE
Harmony Implementation of material Time Picker

### PR DESCRIPTION
**Pull request by sunnky-kr**
Material TIme Picker library with demo file created

### CODE REVIEW

File: Material_TimePicker/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@ohos/material_timepicker",
```
Comment (archanaatr) [inline Material_TimePicker/package.json]: Change the folder name from Material_TimePicker to MaterialTimePicker

Comment (archanaatr) [inline Material_TimePicker/package.json]: Change this name to materialtimepicker

File: Material_TimePicker/src/main/ets/components/TImeModel.ets [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    else {
+      this.hour = hour % 12 + (this.period == ClockPeriod.PM ? 12 : 0)
+    }
+  }
+
+  public setHourOfDay(hour: number) {
+    if (hour < 0 || hour > 24)  return
+    hour %= 24
+    this.period = hour >= 12 ? ClockPeriod.PM : ClockPeriod.AM
```
Comment (archanaatr) [inline Material_TimePicker/src/main/ets/components/TImeModel.ets]: Rename the file name to TimeModel.ets

File: Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  hourButtonClickTime: Date
+  minuteButtonClickTime: Date
+  StateColors: ResourceColor[]= ["#ffd8ddff", "#e4e4e4"]
+  TimeStateColors: ResourceColor[]= ["#ff5e00ff", "#ff282828"]
+  periodBGColors: ResourceColor[]= ["#ffd8ddff", Color.White]
+  periodStateColors: ResourceColor[]= ["#ff5e00ff", "#ffabaaad"]
+  borderColor: ResourceColor= "#ffabaaad"
+
+  aboutToAppear() {
+    if (this.timePickerModel.model.period == ClockPeriod.PM)
```
Comment (archanaatr) [inline Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets]: Add even the single statement of if condition within braces

File: entry/package.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "ohos": {
+    "org": "huawei",
+    "buildTool": "hvigor",
+    "directoryLevel": "module"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "@ohos/material_timepicker": "file:../Material_TimePicker"
```
Comment (archanaatr) [inline entry/package.json]: change to @ohos/materialtimepicker

File: entry/src/main/ets/pages/index.ets [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    .height("100%")
+    .padding({ top: '50vp', left: '20vp' })
+  }
+}
+
+@Component
+struct TimePickerButton {
+  @State time: string= ""
+  width: Length= '100vp'
+  controller: CustomDialogController= new CustomDialogController({
```
Comment (archanaatr) [inline entry/src/main/ets/pages/index.ets:63]: better to open it as a dialog from library or do you want to support full page picker display also?

File: Material_TimePicker/hvigorfile.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline Material_TimePicker/hvigorfile.js]: add copyright text

File: Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        }
+        .width('100%')
+        .height('80vp')
+        .borderRadius('2vp')
+        .borderWidth('2vp')
+        .borderColor(this.timePickerModel.format != ElementFormat.FORMAT_MINUTE && this.timePickerModel.inputMethod == InputMethod.TextInputMethod ? "#ff5e00ff" : this.borderColor)
+        .backgroundColor(this.StateColors[this.timePickerModel.format == ElementFormat.FORMAT_MINUTE?1:0])
+        .onClick(() => {
+          this.timePickerModel.setFormat(this.timePickerModel.model.format == TimeFormat.CLOCK_12H ? ElementFormat.FORMAT_12H : ElementFormat.FORMAT_24H)
+          if (this.hourButtonClickTime == undefined) {
```
Comment (Himanshu-0123) [inline Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets]: handle null check also

File: entry/hvigorfile.js [hunk 8]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline entry/hvigorfile.js:17]: add copyright text

**Comment by Himanshu-0123** [comment]
add copyright text in all js/ets/ts files

File: Material_TimePicker/src/main/ets/components/TimePickerModel.ets [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+  setKeyboardSymbol(keyBoardSymbol: Resource) {
+    this.inputSymbol[0] = keyBoardSymbol
+  }
+
+  setClockSymbol(clockSymbol: Resource) {
+    this.inputSymbol[1] = clockSymbol
+  }
+
+  setInputMethod(method: InputMethod) {
+    if (this.inputMethod != method) {
```
Comment (Himanshu-0123) [inline Material_TimePicker/src/main/ets/components/TimePickerModel.ets]: handle null and undefined validation in all setter methods

File: Material_TimePicker/src/main/ets/components/TImeModel.ets [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+  format: TimeFormat
+  constructor(hour: number, minute: number, format: TimeFormat) {
+    this.format = format
+    this.setHour(hour)
+    this.setMinute(minute)
+    this.setHourOfDay(hour)
+  }
+
+  public setHour(hour: number) {
+    if (hour < 0 || hour > 24)  return
```
Comment (Himanshu-0123) [inline Material_TimePicker/src/main/ets/components/TImeModel.ets]: handle null & undefined validation in all setter methods

File: entry/src/main/ets/pages/index.ets (hunk 5 above)
Comment (sunnky-kr) [reply entry/src/main/ets/pages/index.ets:63]: It is the way it is intended

File: Material_TimePicker/package.json (hunk 1 above)
Comment (sunnky-kr) [reply Material_TimePicker/package.json]: done

Comment (sunnky-kr) [reply Material_TimePicker/package.json]: done

File: Material_TimePicker/src/main/ets/components/TImeModel.ets (hunk 2 above)
Comment (sunnky-kr) [reply Material_TimePicker/src/main/ets/components/TImeModel.ets]: done

File: Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets (hunk 3 above)
Comment (sunnky-kr) [reply Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets]: done

File: entry/package.json (hunk 4 above)
Comment (sunnky-kr) [reply entry/package.json]: done

File: Material_TimePicker/hvigorfile.js (hunk 6 above)
Comment (sunnky-kr) [reply Material_TimePicker/hvigorfile.js]: done

File: Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets (hunk 7 above)
Comment (sunnky-kr) [reply Material_TimePicker/src/main/ets/components/ChipTextInputComboComponent.ets]: done

File: entry/src/main/ets/pages/index.ets (hunk 5 above)
Comment (PatExplores) [reply entry/src/main/ets/pages/index.ets:63]: Dialogs are called in this way only, its proper.

File: entry/hvigorfile.js (hunk 8 above)
Comment (sunnky-kr) [reply entry/hvigorfile.js:17]: done

File: Material_TimePicker/src/main/ets/components/TimePickerModel.ets (hunk 9 above)
Comment (sunnky-kr) [reply Material_TimePicker/src/main/ets/components/TimePickerModel.ets]: done

File: Material_TimePicker/src/main/ets/components/TImeModel.ets (hunk 10 above)
Comment (sunnky-kr) [reply Material_TimePicker/src/main/ets/components/TImeModel.ets]: done

**Pull request merged**